src/Zio/UPathExtensions.cs

```diff
@@ -257,10 +260,11 @@ public static UPath AssertNotNull(this UPath path, string name = "path")
     /// <exception cref="System.ArgumentException">If the path is not absolute using the parameter name from <paramref name="name"/></exception>
     public static UPath AssertAbsolute(this UPath path, string name = "path")
     {
-        AssertNotNull(path, name);
-
         if (!path.IsAbsolute)
-            throw new ArgumentException($"Path `{path}` must be absolute", name);
-        return path.FullName;
+            Throw(path, name);
+
+        return path;
+
+        static void Throw(UPath path, string name) => throw new ArgumentException($"Path `{path}` must be absolute", name);
```

Comment:
In the throw you could check if path is null and throw an ArgumentNullException or keep the ArgumentException for the not absolute case.

Reply:
smart, didn't think of it. I'll make a comment in the code to note why it is implemented as such.